python/lsst/daf/butler/dimensions/_coordinate.py

```diff
@@ -69,17 +76,20 @@
 DataCoordinate or other data ID.
 """
 
+SerializedDataId = dict[str, DataIdValue]
```

Comment:
You mean in this bit?:
```
DataId = DataCoordinate | Mapping[str, Any]
```

Not really sure what that's about.  DataCoordinate proper does seem to use the DataIdValue pretty consistently everywhere.  That `DataId` union exists mainly as a parameter for Butler methods I think, so maybe when you guys added mypy there was too much random code that was passing in like stringifiable things that weren't actually strings or something.  There is a bit of code in `standardize` doing things like coercing numpy.int64 to int, but not as much as I would have thought.

I kind of feel like there are a values that aren't ints or strings though.  Isn't some stuff like floats or datetimes?

I sort of suspect that the existing QuantumGraph code or wherever this is getting used just happens to not use anything unusual, and we may end up with some stuff getting accidentally coerced to strings at some point.  In any case I just pulled out an alias for the existing definition from SerializedDataCoordinate, so a problem for another day.

Reply:
Dimensions do not use floats because you can't do equality on them. Dimensions are strings or int (and as you found, integral types so really should be `numbers.Integral` or something instead of int).